src/pybind/mgr/dashboard/frontend/e2e/pools/pools.po.ts

```diff
@@ -17,86 +17,69 @@ export class PoolPageHelper extends PageHelper {
   }
 
   @PageHelper.restrictTo(pages.index)
-  exist(name: string, oughtToBePresent = true): promise.Promise<any> {
-    return this.getTableCellByContent(name).then((elem) => {
-      const waitFn = oughtToBePresent ? EC.visibilityOf(elem) : EC.invisibilityOf(elem);
-      return browser.wait(waitFn, Helper.TIMEOUT).catch(() => {
-        const visibility = oughtToBePresent ? 'invisible' : 'visible';
-        const msg = `Pool "${name}" is ${visibility}, but should not be. Waiting for a change timed out`;
-        return promise.Promise.reject(msg);
-      });
-    });
+  async exist(name: string, oughtToBePresent = true) {
+    const tableCell = await this.getTableCellByContent(name);
+    const waitFn = oughtToBePresent ? EC.visibilityOf(tableCell) : EC.invisibilityOf(tableCell);
+    try {
+      await browser.wait(waitFn, Helper.TIMEOUT);
+    } catch (e) {
+      const visibility = oughtToBePresent ? 'invisible' : 'visible';
+      const msg = `Pool "${name}" is ${visibility}, but should not be. Waiting for a change timed out`;
+      return Promise.reject(msg);
+    }
+    return Promise.resolve();
   }
 
   @PageHelper.restrictTo(pages.create)
-  create(name: string, placement_groups: number, ...apps: string[]): promise.Promise<any> {
+  async create(name: string, placement_groups: number, ...apps: string[]): Promise<any> {
     const nameInput = $('input[name=name]');
-    nameInput.clear();
+    await nameInput.clear();
     if (!this.isPowerOf2(placement_groups)) {
       return Promise.reject(`Placement groups ${placement_groups} are not a power of 2`);
     }
-    return nameInput.sendKeys(name).then(() => {
-      element(by.cssContainingText('select[name=poolType] option', 'replicated'))
-        .click()
-        .then(() => {
-          expect(element(by.css('select[name=poolType] option:checked')).getText()).toBe(
-            ' replicated '
-          );
-          $('input[name=pgNum]')
-            .sendKeys(protractor.Key.CONTROL, 'a', protractor.Key.NULL, placement_groups)
-            .then(() => {
-              this.setApplications(apps);
-              return element(by.css('cd-submit-button')).click();
-            });
-        });
-    });
+    await nameInput.sendKeys(name);
+    await element(by.cssContainingText('select[name=poolType] option', 'replicated')).click();
+
+    expect(await element(by.css('select[name=poolType] option:checked')).getText()).toBe(
```

Comment:
why does some expects have `expect(await element(by...` and others just have `expect(element(by...`

Reply:
Only promises can be awaited. `getText()`, `click()` and `sendKeys()`, for instance, return promises. If an expect does not contain an `await`, then it's either wrong or the value to be tested is not a promise.

Reply:
Since the `expect` without the inside `await` is currently passing on the tests, I don't think it's necessary to use it there.
But it seems Angular team uses the `await` before the `expect` [[example](https://github.com/angular/protractor/blob/5.4.1/spec/ts/basic/element_spec.ts)], so I would suggesting doing the same:

```suggestion
    await expect(element(by.css('select[name=poolType] option:checked')).getText()).toBe(    
```